.changeset/poor-apes-cheat.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': minor
+---
+
+Adds `limitInputPixels` option for the Sharp image service to be passed on to Sharp.
```

Comment:
``````suggestion
Adds an option for the Sharp image service to allow large images to be processed. Set `limitInputPixels: false` to bypass the default image size limit:

```
// astro.mjs.config
example showing this config
```

``````

(I'm assuming this is a config-wide option set, and not e.g. per-image.)

Depending on the appropriate code sample here, the same will need to be added to the astro types file. Since sharp is the default image service, someone might not have this in their config already, but they'd need to add it in order to pass the option. So, I'm thinking somewhere here: https://docs.astro.build/en/reference/configuration-reference/#imageservice but it would have to account for the fact that someone is only adding this NOT to actually change the image service used, but rather to configure the default one. (Since, this option is JUST for Sharp, and not the other services)

Depending on whatever's decided here, for an actual docs PR, we can even get away with just linking to the config option from here: https://docs.astro.build/en/guides/images/#default-image-service and don't need to explain it. We really should have a link there anyway!  😅 

Sorry this requires a bit of figuring out, but I'll make sure I get up early tomorrow and am happy to help with it!

Reply:
Thanks for the review! Yeah I have a rough idea to improve this now.

Thanks for linking me to the docs. I actually would've have documented something like:

```js
import { defineConfig, sharpImageService } from 'astro/config'

export default defineConfig({
  image: {
    service: sharpImageService({ limitInputPixels: false })
  }
})
```

But I think it's easier to keep the object format for now, so it applies more generally to other kinds of image services.

Reply:
I've updated the changesets and jsdoc. Let me know what you think!